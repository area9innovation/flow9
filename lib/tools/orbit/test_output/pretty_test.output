Flow compiler (3rd generation)

Processing 'orbit' on http server

Testing prettyOrbit function

Expression 1: 10 - 20
Expression 2: x + 2
Variable not found: addNodeToOGraph
Cannot call non-function value: 0
Variable not found: addNodeToOGraph
Cannot call non-function value: 0
ERROR: Expected int, not Subtract(10, 20)
ERROR: Expected int, not Add("x", 2)

Added nodes with IDs: 0, 0

Graph contents:
Variable not found: extractOGraphNode
Cannot call non-function value: 0

Extracted node: 10 - 20

Done!
Result: "\nDone!"
