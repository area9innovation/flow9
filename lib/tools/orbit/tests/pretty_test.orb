// Test the prettyOrbit function

// Using the ast annotation, we quote the argument
fn quote(a : ast) = a;

println("Testing prettyOrbit function");

// Create some expressions
let expr1 = quote(10 - 20);
let expr2 = quote(x + 2);

// Print them using prettyOrbit
println("\nExpression 1: " + prettyOrbit(expr1));
println("Expression 2: " + prettyOrbit(expr2));

// Create a graph
let graph = makeOGraph("pretty_test");

// Add nodes to the graph
let nodeId1 = addNodeToOGraph(graph, "math", expr1);
let nodeId2 = addNodeToOGraph(graph, "math", expr2);

println("\nAdded nodes with IDs: " + i2s(nodeId1) + ", " + i2s(nodeId2));

// Print the graph
println("\nGraph contents:");
printOGraph(graph);

// Extract a node and print it
let extracted = extractOGraphNode(graph, nodeId1);
println("\nExtracted node: " + prettyOrbit(extracted));

println("\nDone!")