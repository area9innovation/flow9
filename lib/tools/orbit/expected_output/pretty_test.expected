Flow compiler (3rd generation)

Processing 'orbit' on http server

Testing prettyOrbit function

Expression 1: 10 - 20
Expression 2: x + 2

Added nodes with IDs: 1, 2

Graph contents:

Extracted node: "expr1"

Done!
Result: "\nDone!"
VERIFY - THIS OUTPUT NEEDS HUMAN VERIFICATION
